shared/js/background/chrome-events.es6.js

```diff
@@ -214,6 +279,44 @@ chrome.runtime.onMessage.addListener((req, sender, res) => {
         res(pixel.fire.apply(null, fireArgs))
         return true
     }
+
+    if (req.checkThirdParty) {
+        const action = {
+            isThirdParty: false,
+            shouldBlock: false,
+            tabRegisteredDomain: null,
+            isTrackerFrame: false,
+            policy: cookieConfig.firstPartyCookiePolicy
+        }
+        if (chrome.runtime.lastError) { // Prevent thrown errors when the frame disappears
+            return true
+        }
+        if (blockingExperimentActive()) {
+            const tab = tabManager.get({ tabId: sender.tab.id })
+            // determine the register domain of the sending tab
+            const tabUrl = tab ? tab.url : sender.tab.url
+            const parsed = tldts.parse(tabUrl)
+            action.tabRegisteredDomain = parsed.isIp ? parsed.hostname : parsed.domain || parsed.hostname
```

Comment:
in what cases url is not an IP but doesn't have domain? (why we have to do `|| parsed.hostname` part?)

Reply:
`localhost` and any other hostname without a `.`.